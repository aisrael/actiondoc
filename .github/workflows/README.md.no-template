Generate GitHub Action documentation
====

Generates documentation from your GitHub Action's `action.yml`

## Inputs

| Name               | Description                                                                             | Required | Default |
|--------------------|-----------------------------------------------------------------------------------------|----------|---------|
| template-filename  | The path to the ERB template to use to generate the documentation.                      | No       |         |
| output-filename    | The filename to save the output to. If not specified, simply prints to standard output. | No       |         |
| path-to-action-yml | The path to the `action.yml` file to generate documentation for.                        | No       |         |
| working-directory  | The directory to perform the action in, if not $GITHUB_WORKSPACE                        | No       |         |
