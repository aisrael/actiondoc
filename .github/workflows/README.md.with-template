actiondoc
====

Generates documentation from your GitHub Action's `action.yml`

## Inputs

| Name               | Description                                                      | Required | Default |
|--------------------|------------------------------------------------------------------|----------|---------|
| path-to-action-yml | The path to the `action.yml` file to generate documentation for. | No       |         |
| working-directory  | The directory to perform the action in, if not $GITHUB_WORKSPACE | No       |         |

See [Basic Usage](features/basic_usage.feature) for some basic usage.

### Example

Given an `action.yml` that contains:

```yaml
name: actiondoc
  description: >-
    Generates documentation from your GitHub Action's `action.yml`

  inputs:
    path-to-action-yml:
      description: >-
        The path to the `action.yml` file to generate documentation for.
        required: false
```

Then when running `actiondoc` it should output

```markdown
actiondoc
====

Generates documentation from your GitHub Action's `action.yml`

## Inputs

| Name               | Description                                                      | Required | Default |
|--------------------|------------------------------------------------------------------|----------|---------|
| path-to-action-yml | The path to the `action.yml` file to generate documentation for. | No       |         |
```
