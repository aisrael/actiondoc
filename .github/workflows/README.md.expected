Generate GitHub Action documentation
====

Generates documentation from your GitHub Action's `action.yml`

## Inputs

| Name               | Description                                                      | Required | Default |
|--------------------|------------------------------------------------------------------|----------|---------|
| path-to-action-yml | The path to the `action.yml` file to generate documentation for. | No       |         |
| working-directory  | The directory to perform the action in, if not $GITHUB_WORKSPACE | No       |         |
